samples/contrib/aws-samples/debugger_component_demo/debugger-component-demo.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+# Uncomment the apply(use_aws_secret()) below if you are not using OIDC
+# more info : https://github.com/kubeflow/pipelines/tree/master/samples/contrib/aws-samples/README.md
+
+import kfp
+import json
+import os
+import copy
+from kfp import components
+from kfp import dsl
+from kfp.aws import use_aws_secret
+
+
+cur_file_dir = os.path.dirname(__file__)
+components_dir = os.path.join(cur_file_dir, '../../../../components/aws/sagemaker/')
+
+sagemaker_train_op = components.load_component_from_file(components_dir + '/train/component.yaml')
+
+debug_hook = {
+    'S3OutputPath':'s3://kubeflow-pipeline-data/mnist_kmeans_example/hook_config'
+}
+
+debug_hook['CollectionConfigurations'] = []
+
+collection_list = {
+    'feature_importance' : {
+        'save_interval': '5'
+    }, 
+    'losses' : {
+        'save_interval': '10'
+    },
+    'average_shap': {
+        'save_interval': '5'
+    },
+    'metrics': {
+        'save_interval': '5'
+    }
+}
+
+for key, val in collection_list.items():
+    debug_hook['CollectionConfigurations'].append({'CollectionName': key, 'CollectionParameters': val})
+
+loss_rule = {
+    'RuleConfigurationName': 'LossNotDecreasing',
+    'RuleEvaluatorImage': '503895931360.dkr.ecr.us-east-1.amazonaws.com/sagemaker-debugger-rules:latest',
+    'RuleParameters': {
+        'rule_to_invoke': 'LossNotDecreasing',
+        'tensor_regex': '.*'
+    }
+}
+
+overtraining_rule = {
+    'RuleConfigurationName': 'Overtraining',
+    'RuleEvaluatorImage': '503895931360.dkr.ecr.us-east-1.amazonaws.com/sagemaker-debugger-rules:latest',
+    'RuleParameters': {
+        'rule_to_invoke': 'Overtraining',
+        'patience_train': '10',
+        'patience_validation': '20'
+    }
+}
+
+debug_rule_configurations=[loss_rule, overtraining_rule]
+
+bad_hyperparameters = {
+    'objective': 'reg:squarederror',
+    'max_depth': '5', 
+    'eta': '0', 
+    'gamma': '4', 
+    'min_child_weight': '6', 
+    'silent': '0', 
+    'subsample': '0.7', 
+    'num_round': '50'
+}
+
+channelObjList = []
+
+channelObj = {
+    'ChannelName': '',
+    'DataSource': {
+        'S3DataSource': {
+            'S3Uri': '',
+            'S3DataType': 'S3Prefix',
+            'S3DataDistributionType': 'FullyReplicated'
+        }
+    },
+    'ContentType': 'text/csv',
+    'CompressionType': 'None',
+    'RecordWrapperType': 'None',
+    'InputMode': 'File'
+}
+
+channelObj['ChannelName'] = 'train'
+channelObj['DataSource']['S3DataSource']['S3Uri'] = 's3://kubeflow-pipeline-data/mnist_kmeans_example/input/valid_data.csv'
+channelObjList.append(copy.deepcopy(channelObj))
```

Comment:
See [the mnist sample](https://github.com/kubeflow/pipelines/blob/master/samples/contrib/aws-samples/mnist-kmeans-sagemaker/mnist-classification-pipeline.py) for a more elegant solution for these objects.